kv/txn_coord_sender.go

```diff
@@ -729,19 +732,10 @@ func (tc *TxnCoordSender) updateState(ctx context.Context, ba roachpb.BatchReque
 	case *roachpb.OpRequiresTxnError:
 		panic("OpRequiresTxnError must not happen at this level")
 	case *roachpb.ReadWithinUncertaintyIntervalError:
-		// Mark the host as certain. See the protobuf comment for
-		// Transaction.CertainNodes for details.
-		if t.NodeID == 0 {
-			panic("no replica set in header on uncertainty restart")
-		}
-		// No more restarts for this node for anything after ExistingTimestamp.
-		newTxn.UpdateObservedTimestamp(t.NodeID, t.ExistingTimestamp)
 		// If the reader encountered a newer write within the uncertainty
-		// interval, move the timestamp forward, just past that write or
-		// up to ObservedTimestamp, whichever comes first.
-		candidateTS := newTxn.ObservedTimestamp
-		candidateTS.Backward(t.ExistingTimestamp.Add(0, 1))
-		newTxn.Timestamp.Forward(candidateTS)
+		// interval, we simply move past the last observed timestamp from the
```

Comment:
s/we simply move past the last observed timestamp from the node./advance the txn's timestamp just past the last observed timestamp from the node./